Applications/Spire/Source/Ui/CalendarDatePicker.cpp

```diff
@@ -307,8 +322,10 @@ CalendarDatePicker::CalendarDatePicker(
     set(BackgroundColor(QColor(0x4B23A0))).
     set(border(0, QColor(Qt::transparent))).
     set(TextColor(QColor(0xFFFFFF)));
-  set_style(*calendar_view, std::move(calendar_style));
-  layout->addWidget(calendar_view);
+  set_style(*m_calendar_view, std::move(calendar_style));
+  setFocusProxy(m_calendar_view);
+  setFixedWidth(scale_width(176));
```

Comment:
I think the width can be derived from the width of m_calendar_view (168px) and the horizontal margins (4px for left and 4px for right).